bin/release-to-branch.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+set -e
+
+CURRENT_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
+BRANCH_NAME="pre-release-$CURRENT_BRANCH_NAME"
+
+# Remove local branch is it already exists
+if [ `git branch --list $BRANCH_NAME` ]; then
+   echo "Deleting branch $BRANCH_NAME that already exists."
+   git branch -D $BRANCH_NAME
+   git push origin --delete $BRANCH_NAME
+fi
+
+# Build the package as normal
+npm run build:package
+
+git add package/
+
+git commit -m "Release GOV.UK Frontend to Github branch testing"
+
+# Create a local branch containing the package directory
+echo "Split package/ directory commits into `$BRANCH_NAME` branch..."
+git subtree split --prefix package -b $BRANCH_NAME
+
+# Force the push of the branch to the remote Github origin
+git push origin $BRANCH_NAME:$BRANCH_NAME
+
+# Remove build commit to bring the original branch back to where it was pre-release
+git reset HEAD^ --hard
```

Comment:
If you ran this script with uncommitted changes, you would currently lose those changes.

Running `git subtree split` should have created a new branch, could we just `git checkout CURRENT_BRANCH_NAME` to go back to the original branch, rather than resetting?

It might also be good to detect uncommitted changes and warn the user?

Reply:
We are resetting the build commit before we split rather than resetting the split, so not sure checking the original branch would work?

Agreed with uncommitted stuff...

Reply:
Gotcha, that makes sense. Maybe the script just refuse to run if there are uncommitted changes?

Reply:
I believe this is now fixed? I'm seeing the following if I have uncommitted changes:
```
⚠️ You have unstaged files, please commit these and then try again.
```